File: 2221.find-triangular-sum-of-an-array/solution.go
func triangularSum(nums []int) int {
    for n := len(nums); n > 1; n-- {
        for i := 0; i < n - 1; i++ {
            nums[i] = (nums[i] + nums[i + 1]) % 10
        }
    }
    return nums[0]
}

File: 2221.find-triangular-sum-of-an-array/solution.kt
class Solution {
    fun triangularSum(nums: IntArray): Int = rec(nums.asList())

    private tailrec fun rec(nums: List<Int>): Int =
        if (nums.size == 1) {
            nums.first()
        } else {
            rec(
                nums
                    .zipWithNext()
                    .map { (a, b) -> (a + b) % 10 }
                )
        }
}